Updating build github actions

mcottontensor opened the pull request:
Updating the workflows.
- Moved the node version to a variable so it's easier to see/change. Probably wants to be a repo variable.
- Added a frontend build action to test that the frontend components build after any change
- Build checks run on any branch, not just master.

lukehb commented on Frontend/README.md:
```diff
@@ -39,3 +39,255 @@ If you are developing your implementation based on the library, the process is s
+
+## Accessing the Pixel Streaming Blueprint API
+
+The Pixel Streaming Plugin that runs within the Unreal Engine exposes a Blueprint API that you can use in your gameplay logic to handle custom UI events sent by the player HTML page, and to emit events from the Unreal Engine to the player page.
```
This section needs to stay on the UE docs side for now, we can think about where it the best home for it.

lukehb commented on Frontend/README.md:
```diff
@@ -39,3 +39,255 @@ If you are developing your implementation based on the library, the process is s
 
 Copyright &copy; 2022, Epic Games. Licensed under the MIT License, see the file [LICENSE](./LICENSE) for details.
+
+
```
Let's split each of these sections out into their own .md file under the docs and have this Frontend/README.md link to them as a table of contents - like the UE docs.

lukehb commented on Frontend/README.md:
```diff
@@ -39,3 +39,255 @@ If you are developing your implementation based on the library, the process is s
+![PixelStreamingCustomPlayer](Docs\Resources\Images\pixelstreaming-custom-player.JPG)
+
+
+## HTML Page Requirements
```
All sections (except the first one) need something in bold like **TODO: Update this information for the next frontend** and then we can open a corresponding issue too.

lukehb reviewed (changes requested):
Some changes I have left in the comments.

Additionally, let's remove the Mclaren images from the commit.

lukehb commented:
@MWillWallT Landing this, can you open an issue and assign @Belchy06 to update these markdown ToDos some time before 5.2 (e.g. add 5.2 milestone)

mcottontensor committed:
Adding an action to build the frontend components on pushes or pull
requests.
Added an env variable to applicable actions to control the node version
being used. This should probably move to a repo variable eventually.

mcottontensor committed:
Running tests on any branch

lukehb reviewed (approved):
lgtm!

Yeah in 5.5 with /common folder maybe we can add a .NodeVersion file or something like that can be read by cli scripts and actions.
